baseapp-ratings/baseapp_ratings/models.py

```diff
@@ -0,0 +1,94 @@
+import swapper
+from baseapp_core.graphql import RelayModel
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
+from django.contrib.contenttypes.models import ContentType
+from django.db import models, transaction
+
+from django.utils.translation import gettext_lazy as _
+from model_utils.models import TimeStampedModel
+
+
+def update_ratings_indicators(target):
+    if not target:
+        return
+
+    RateModel = swapper.load_model("baseapp_ratings", "Rate")
+    target_content_type = ContentType.objects.get_for_model(target)
+    qs = RateModel.objects.filter(
+        target_content_type=target_content_type,
+        target_object_id=target.pk,
+    )
+    with transaction.atomic():
+        target.ratings_count = qs.count()
+        target.ratings_sum = qs.aggregate(models.Sum("value"))["value__sum"]
+        target.ratings_average = (
+            target.ratings_sum / target.ratings_count if target.ratings_count else 0
+        )
+
+        target.save(update_fields=["ratings_count", "ratings_sum", "ratings_average"])
+
+
+class AbstractBaseRate(TimeStampedModel, RelayModel):
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        related_name="ratings",
+        on_delete=models.CASCADE,
+    )
+
+    target_content_type = models.ForeignKey(
+        ContentType,
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        db_index=True,
+    )
+    target_object_id = models.PositiveIntegerField(blank=True, null=True, db_index=True)
```

Comment:
I think we can remove the `db_index` here as well, I see no reason to filter only by `target_object_id` and not use `target_content_type` in the filter